src/lib/components/CopyCode/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Element that renders copyable code.
+ */
+
+import {BaseElement} from "../BaseElement";
+
+/**
+ * Renders code block that can easily be copied.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class CopyCode extends BaseElement {
+  static get properties() {
+    return {
+      code: {type: String},
+    };
+  }
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute("tabindex", "0");
+
+    if (this.firstChild) {
+      const copyButton = document.createElement("input");
+      copyButton.setAttribute("type", "button");
+      copyButton.setAttribute("tabindex", "0");
+      copyButton.className = "w-copy-code-button";
+
+      this.firstChild.prepend(copyButton);
+
+      this.addEventListener("click", (e) => {
+        if (e.target === copyButton) {
+          e.preventDefault();
+          this.copyCode();
+        }
+      });
+    }
+  }
+
+  copyCode() {
+    const textarea = document.createElement("textarea");
```

Comment:
Another way to go about this might be to select the text of the firstChild itself. Taking a look at [this doc](https://developers.google.com/web/updates/2015/04/cut-and-copy-commands) I tried:

```
window.getSelection().removeAllRanges();
const range = document.createRange();
range.selectNode(this.firstChild);
window.getSelection().addRange(range);
document.execCommand("copy");
window.getSelection().removeAllRanges();
```

Reply:
I'm digging this solution.